File: fix_main.py
with open('app/src/main/java/com/example/MainActivity.kt', 'r') as f:
    c = f.read()
c = c.replace('CompositionLocalProvider(LocalLanguage provides selectedLanguage) {', 
'''var selectedLanguage by remember { mutableStateOf(sharedPrefs.getString("selected_language", "es") ?: "es") }
    CompositionLocalProvider(LocalLanguage provides selectedLanguage) {''')
with open('app/src/main/java/com/example/MainActivity.kt', 'w') as f:
    f.write(c)

File: fix_main.js
const fs = require('fs');
let code = fs.readFileSync('app/src/main/java/com/example/MainActivity.kt', 'utf8');

code = code.replace(/\/\/\s*Ambient Runic Particles Floating across Bottom Navigation Bar(?: in background)?\n\s*RunicNavBarParticleAnimation\(\n\s*particleCount = 1[46],\n\s*accentColor = .*?\n\s*\)/g, 
function(match) {
    return 'if (com.example.ui.theme.AppThemeManager.isParticlesEnabled) {\n                ' + match + '\n                }';
});

fs.writeFileSync('app/src/main/java/com/example/MainActivity.kt', code);
console.log("MainActivity replaced");
